src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,108 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int SIZE = 10;
+
+    private int filledSize;
+    private T[] array;
+
+    public ArrayList() {
+        array = (T[]) new Object[SIZE];
+        filledSize = 0;
+    }
+
+    public ArrayList(int incomeSize) {
+        array = (T[]) new Object[incomeSize];
+        filledSize = 0;
+    }
+
+    public void sizeCheck() {
+        if (filledSize == array.length) {
+            T[] newArray = (T[]) new Object[(int) (array.length * 1.5)];
+            System.arraycopy(array, 0, newArray, 0, filledSize);
+            array = newArray;
+        }
+    }
 
     @Override
     public void add(T value) {
-
+        sizeCheck();
+        array[filledSize] = value;
+        filledSize++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > filledSize) {
+            throw new ArrayIndexOutOfBoundsException();
+        } else {
```

Comment:
we don't need `else` here, because our code execution will stop if we enter `if` block
also, please make the exception more informative